src/theme/ThemeProvider.tsx

```diff
@@ -24,20 +25,18 @@ const missingProvider = () => {
   throw new Error('ThemeProvider is missing')
 }
 
-const useSavedColorScheme = (options: UseQueryOptions<ColorScheme | null> = {}) => {
-  const query = useQuery({
+const useSavedColorScheme = ({initialData}: {initialData?: ColorScheme}) => {
+  const query = useQuery<ColorScheme | null>({
+    initialData,
```

Comment:
this might briefly display the initialData before switching to the async loaded theme